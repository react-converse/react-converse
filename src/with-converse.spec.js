import React from 'react';
import { mount } from 'enzyme';
import { withConverse } from '~/index';

// https://github.com/airbnb/enzyme/issues/1509#issuecomment-401207909
jest.mock('react', () => {
  const originalReact = require.requireActual('react');
  return {
    ...originalReact,
    createContext: require.requireActual('create-react-context').default,
  };
});

describe('withConverse', () => {
  it('injects API props', () => {
    const ComponentMock = jest.fn().mockReturnValue(<div />);
    const TestComponent = withConverse(ComponentMock);
    mount(<TestComponent />);
    expect(ComponentMock).toHaveBeenCalledWith(
      expect.objectContaining({
        converse: expect.any(Object),
      }),
      expect.any(Object)
    );
  });

  it('wraps `displayName`', () => {
    const ComponentMock = () => <div />;
    ComponentMock.displayName = 'Mock';
    const TestComponent = withConverse(ComponentMock);
    expect(TestComponent.displayName).toMatchSnapshot();
  });
});
