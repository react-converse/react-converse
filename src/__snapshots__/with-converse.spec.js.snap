// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withConverse wraps \`displayName\` 1`] = `"withConverse(Mock)"`;
